### PR TITLE
implement http sink

### PR DESCRIPTION
**Pull request by almonteb**
Implements an interface that allows metrics to be sent via HTTP in the following JSON format:

```
{
    "metrics": [
        {"key": "foo", "value": "123", "timestamp": "1500000000"},
        {"key": "bar", "value": "456", "timestamp": "1500000001"},
    ]
}
```

### CODE REVIEW

**Comment by johnkeates** [comment]
This currently fails the CI test, probably some missing module?

File: integ/test_http.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""
+Testing the http sink
+"""
+import json
+from mock import patch
+
```
Comment (johnkeates) [inline integ/test_http.py]: Maybe something like a HTTP or Request import is needed here?

Comment (almonteb) [inline integ/test_http.py]: The python instance that runs the tests didn't have requests installed. Added the lib to .travis.yml

**Comment by johnkeates** [comment]
Looks like it tests fine now. Not sure if manual testing is broken as well, also not sure where to put requests==2.18.1 as a dependency. Maybe in the docs?

**Comment by johnkeates** [comment]
Looks like integration testing isn't guaranteed anywhere, we should probably document it and it's requirements somewhere.

**Comment by almonteb** [comment]
Thanks for merging @johnkeates! 

Looks like INSTALL.md mentions the requirement of pytest on L99 & L118. Would you like me update + PR?

**Comment by johnkeates** [comment]
That would be great!

**Pull request merged**